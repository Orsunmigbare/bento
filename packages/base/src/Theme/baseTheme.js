import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
