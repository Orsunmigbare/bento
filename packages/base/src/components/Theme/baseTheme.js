export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
