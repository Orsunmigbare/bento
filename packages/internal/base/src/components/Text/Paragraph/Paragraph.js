import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { fontFamily, themify } from "@staccx/theme"

const Paragraph = ({ level, children, className, ...restProps }) => {
  return (
    <P className={className} {...restProps}>
      {children}
    </P>
  )
}

export const PARAGRAPH = "paragraph"

export const P = styled.p`
  font-family: ${fontFamily.body};
  ${themify(PARAGRAPH)};
`

Paragraph.defaultProps = {
  className: null
}

Paragraph.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  className: PropTypes.string
}

export default Paragraph
