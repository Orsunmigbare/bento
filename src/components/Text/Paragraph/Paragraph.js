import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  fontFamily,
  font,
  applyVariants,
  commonPropTypes,
  themePropTypes
} from "../../../theming"

const Paragraph = ({ children, className, ...restProps }) => {
  return (
    <P className={className} {...restProps}>
      {children}
    </P>
  )
}

Paragraph.themeProps = {
  paragraph: {
    name: "paragraph",
    description: "",
    type: themePropTypes.style
  }
}

export const P = styled.p`
  font-family: ${fontFamily.body};
  line-height: ${font.lineHeight};
  ${applyVariants(Paragraph.themeProps.paragraph)};
`

Paragraph.defaultProps = {
  className: null
}

Paragraph.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string
}

export default Paragraph
