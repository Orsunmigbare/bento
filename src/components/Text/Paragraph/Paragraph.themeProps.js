import { themePropTypes, createVariantsFunctionFactory } from "../../../theming"

const themeProps = {
  paragraph: {
    name: "paragraph",
    description: "",
    type: themePropTypes.style
  }
}

export default createVariantsFunctionFactory(themeProps)
