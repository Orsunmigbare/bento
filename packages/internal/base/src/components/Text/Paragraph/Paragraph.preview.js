import React from "react"
import Paragraph from "./Paragraph"

const preview = {
  title: "Paragraph",
  category: "Components/Texts",
  tags: ["typography"],
  component: Paragraph,
  render: props => <Paragraph {...props}>This is a Paragraph</Paragraph>
}

export default preview
