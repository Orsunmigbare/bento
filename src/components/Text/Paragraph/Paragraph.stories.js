import React from "react"
import { storiesOf } from "@storybook/react"
import Paragraph from "./Paragraph"

storiesOf("Components|Base/Text/Paragraph", module)
  .add("no props", () => <Paragraph>This is a paragraph</Paragraph>)
  .add("with className", () => (
    <Paragraph className={"Test"}>This is a paragraph</Paragraph>
  ))
