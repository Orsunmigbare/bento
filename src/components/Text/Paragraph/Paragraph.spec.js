import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Paragraph from "./Paragraph"

describe("Paragraph", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Paragraph>Hei</Paragraph>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("className", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Paragraph className="className">Hei</Paragraph>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("No Children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Paragraph />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
