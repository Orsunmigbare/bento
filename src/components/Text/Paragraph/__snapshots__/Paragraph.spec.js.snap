// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Snapshots No Children 1`] = `
<p
  className="sc-AykKC ecIodL"
/>
`;

exports[`Paragraph Snapshots No props 1`] = `
<p
  className="sc-AykKC ecIodL"
>
  Hei
</p>
`;

exports[`Paragraph Snapshots className 1`] = `
<p
  className="sc-AykKC ecIodL className"
>
  Hei
</p>
`;
